contracts/cw20-base/src/contract.rs

```diff
@@ -228,6 +233,95 @@ pub fn execute(
     }
 }
 
+pub fn execute_cw20_to_bank(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    amount: Uint128,
+) -> Result<Response, ContractError> {
+    if amount == Uint128::zero() {
+        return Err(ContractError::InvalidZeroAmount {});
+    }
+    // Verify Cw20 balance of the sender
+    let balance = query_balance(deps.as_ref(), info.sender.to_string())
+        .unwrap()
+        .balance;
+    if amount > balance {
+        return Err(ContractError::InsufficientBalance {});
+    }
+
+    // Burn cw20 tokens of the user
+    // lower balance
+    BALANCES.update(
+        deps.storage,
+        &info.sender,
+        |balance: Option<Uint128>| -> StdResult<_> {
+            Ok(balance.unwrap_or_default().checked_sub(amount)?)
+        },
+    )?;
+    // reduce total_supply
+    TOKEN_INFO.update(deps.storage, |mut info| -> StdResult<_> {
+        info.total_supply = info.total_supply.checked_sub(amount)?;
+        Ok(info)
+    })?;
+
+    // transfer the bank denom to user address from CW20 contract address.
+    // Create and submit BankTransfer sub message. Here sender is the contract address.
+    let config = TOKEN_INFO.load(deps.storage)?;
+
+    let bank_transfer = BankMsg::Send {
```

Comment:
This assumes the contract starts with large (unbounded?) amount of such bank_denom.
That is not the usual case and I don't believe it fits in cw-plus.

I would encourage this to exist in cw-tokens repo, which is more open to community usage of cw20, not the basic standard.